front/lib/auth.ts

```diff
@@ -358,6 +364,62 @@ export class Authenticator {
     });
   }
 
+  /**
+   * Exchanges an Authenticator associated with a system key for one associated with a user.
+   *
+   * /!\ This function should only be used with Authenticators that are associated with a system key.
+   *
+   * @param auth
+   * @param param1
+   * @returns
+   */
+  async exchangeSystemKeyForUserAuthByEmail(
+    auth: Authenticator,
+    { userEmail }: { userEmail: string }
+  ): Promise<Authenticator | null> {
+    if (!auth.isSystemKey()) {
```

Comment:
Note: this will 500. I think it's fine though.

Reply:
Maybe return `null` is better?

Reply:
Right but a Result would be a bit cleaner. But since you don't catch it I'm fine with a throw